Userland/Utilities/tar.cpp

```diff
@@ -33,12 +34,12 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     bool extract = false;
     bool list = false;
     bool verbose = false;
-    bool gzip = false;
-    bool lzma = false;
-    bool xz = false;
+    SetOnce gzip;
+    SetOnce lzma;
+    SetOnce xz;
     bool no_auto_compress = false;
     StringView archive_file;
-    bool dereference;
```

Comment:
As another evidence on why raw boolean flags could be bad... this is an uninitialized variable (which I assumed the compiler will yell at us?). Apparently there are more cases like this, and I fixed this case right now, but I intend on fixing more in the future.